torch/nn/quantized/dynamic/modules/rnn.py

```diff
@@ -203,38 +235,56 @@ def from_float(cls, mod):
         qRNNBase._all_weights = []
         packed_weights = []
         quantized_weights = []
+        orig_weights = []
         for layer in range(qRNNBase.num_layers):
             for direction in range(num_directions):
                 layer_input_size = qRNNBase.input_size if layer == 0 else qRNNBase.hidden_size * num_directions
 
-                def process_weights(ihhh, layer, suffix):
+                def process_weights(ihhh, layer, suffix, dtype):
                     weight_name = 'weight_{}_l{}{}'.format(ihhh, layer, suffix)
                     bias_name = 'bias_{}_l{}{}'.format(ihhh, layer, suffix)
 
                     weight = getattr(mod, weight_name)
                     bias = getattr(mod, bias_name)
-                    # for each layer, for each direction we need to quantize and pack
-                    # weights and pack parameters in this order:
-                    #
-                    #   w_ih, w_hh, b_ih, b_hh
-                    weight_observer(weight)
-                    wt_scale, wt_zp = weight_observer.calculate_qparams()
-                    qweight = torch.quantize_linear(
-                        weight.float(), float(wt_scale), int(wt_zp), torch.qint8)
-                    packed_weight = \
-                        torch.ops.quantized.linear_prepack(qweight, bias)
-
-                    params = [packed_weight, bias]
-                    pos_names = ['w', 'b']
-                    ret_name = ['{}_{}_l{}{}'.format(
-                        name, ihhh, layer, suffix) for name in pos_names]
-                    quantized_weights.append(qweight)
-                    packed_weights.append(ret_name[0])
-                    return params, ret_name
+
+                    if dtype == torch.qint8:
```

Comment:
Same comments as above

BTW we should probably deduplicate this code between from_float and the constructor

Reply:
Have rebased it. Yes: we should consider a way to deduplicate the code between `from_float` and `__init__`.